job_creator.php

```diff
@@ -132,10 +132,24 @@ public function getInstallerVersion(
 
     public function createJob(int $phpIndex, array $opts): array
     {
+        $isInstallerModule = in_array($this->repoName, INSTALLER_MODULES);
+        $isPreferLoweset = strpos($opts['composer_args'] ?? '', '--prefer-lowest') !== false;
+        $installerVersion = $this->installerVersion;
+        $cmsMajor = BranchLogic::getCmsMajor($this->repoData, $this->branch, $this->getComposerJsonContent());
+        if ($cmsMajor >= 5 && !$isInstallerModule && $isPreferLoweset && $installerVersion) {
+            // --prefer-lowest jobs should use the lowest installer version for non-installer modules
+            // gha-ci will increment the installer version higher if composer is unable to install
+            // and try installing again
```

Comment:
For prefer-lowest it probably shouldn't increment the installer version - that goes against the entire point of using prefer-lowest.

Reply:
Incrementing installer is the entire point of this solution

It always starts off trying to using installer 5.0. If that fails to installs, then it attempts installer 5.1

This would have caught out the [original issue](https://github.com/silverstripe/silverstripe-elemental/issues/1192) that started all of this.  That version of elemental has a framework dep of ^5, instead of ^5.1.  What happened the --prefer-lowest builds were installing using a recent version of installer that included a version of framework that DID have the eagerLoad API (added in 5.1).  However with this new solution it would have installed with installer 5.0, and this would have lead to the CI build going red and we'd see the issue.

With this solution, which the framework dep correctly being ^5.1 instead, it will fail to install with installer 5.0, then it will successfully install with installer 5.1. Which is what we want to test --prefer-lowest with because that's what the specified minimum version of framework/installer is.